From 749fbee32eeb91a511c65b78aec55e10bb46d00a Mon Sep 17 00:00:00 2001
From: Jozef Mlich <jmlich83@gmail.com>
Date: Tue, 25 May 2021 17:50:38 +0200
Subject: [PATCH] Empty implementation of some functions:

bool VoiceCallHandler::isReady() const
bool OfonoVoiceCallProvider::setPoweredAndOnline(bool)
void CallChannelHandler::onCallChannelCallLocalHoldStateChanged(Tp::LocalHoldState state,Tp::LocalHoldStateReason reason)

upstream tracker https://git.sailfishos.org/mer-core/voicecall/issues/1
---
 plugins/declarative/src/voicecallhandler.cpp           | 6 ++++++
 plugins/providers/ofono/src/ofonovoicecallprovider.cpp | 5 +++++
 plugins/providers/telepathy/src/callchannelhandler.cpp | 5 +++++
 3 files changed, 16 insertions(+)

diff --git a/plugins/declarative/src/voicecallhandler.cpp b/plugins/declarative/src/voicecallhandler.cpp
index f203576..79a35e0 100644
--- a/plugins/declarative/src/voicecallhandler.cpp
+++ b/plugins/declarative/src/voicecallhandler.cpp
@@ -452,3 +452,9 @@ void VoiceCallHandler::onPendingCallFinished(QDBusPendingCallWatcher *watcher)
         DEBUG_T("Received successful reply for member: %s", qPrintable(reply.reply().member()));
     }
 }
+
+bool VoiceCallHandler::isReady() const
+{
+    qWarning() << "TODO: missing implementation";
+    return false;
+}
diff --git a/plugins/providers/ofono/src/ofonovoicecallprovider.cpp b/plugins/providers/ofono/src/ofonovoicecallprovider.cpp
index 4ae8b57..fc8d723 100644
--- a/plugins/providers/ofono/src/ofonovoicecallprovider.cpp
+++ b/plugins/providers/ofono/src/ofonovoicecallprovider.cpp
@@ -224,3 +224,8 @@ void OfonoVoiceCallProvider::onCallRemoved(const QString &call)
     emit this->voiceCallRemoved(handlerId);
     emit this->voiceCallsChanged();
 }
+
+bool OfonoVoiceCallProvider::setPoweredAndOnline(bool) {
+    qWarning() << "TODO: missing implementation";
+    return false;
+}
\ No newline at end of file
diff --git a/plugins/providers/telepathy/src/callchannelhandler.cpp b/plugins/providers/telepathy/src/callchannelhandler.cpp
index aff299b..fb04b6a 100644
--- a/plugins/providers/telepathy/src/callchannelhandler.cpp
+++ b/plugins/providers/telepathy/src/callchannelhandler.cpp
@@ -517,3 +517,8 @@ void CallChannelHandler::setStatus(VoiceCallStatus newStatus)
     d->status = newStatus;
     emit statusChanged(d->status);
 }
+
+
+void CallChannelHandler::onCallChannelCallLocalHoldStateChanged(Tp::LocalHoldState state,Tp::LocalHoldStateReason reason) {
+    qWarning() << "TODO missing implementation";
+}
\ No newline at end of file
-- 
2.31.1

